### DISCUSSION TITLE
Media picker 3 show video frame/preview when video picked

### DISCUSSION THREAD

**Discussion by martinlcp** [umbraco/Umbraco-CMS#13312] [Features and ideas]
Can we have a poster frame view of a video picked in Media Picker 3, instead of the icon?

**Comment by nul800sebastiaan** [comment]
What does that mean "poster frame"? Any examples? Do we have this in the media library of Umbraco already or elsewhere in Umbraco?

**Reply by martinlcp** [reply]
So instead of picking this...
![video1](https://user-images.githubusercontent.com/883638/198543845-50a1ef52-2cda-46d2-a7b5-fb907c54c770.png)
you pick this...
![video2](https://user-images.githubusercontent.com/883638/198543885-221ccf7d-82dd-4caa-94ba-ed73f674bb30.png)
Rather than attempting to grab a single frame from actual video file (or squashing the paused video into the thumbnail), provide a "poster frame" in the media type which can also be used as a poster="/posterFile.png" attribute in a view and the thumbnail in the picker.
![video3](https://user-images.githubusercontent.com/883638/198544374-9767e1cb-1553-4ac6-a9ad-d35b4b70573a.png)

**Reply by bjarnef** [reply]
Maybe Umbraco could have a special alias like `umbracoFilePoster`, `umbracoCover` or something else? 

E.g. added to Video media type using a file upload as well:

![image](https://user-images.githubusercontent.com/2919859/198567875-554638b1-abfc-4f7c-88b7-171025e7b4c4.png)

Not sure if this should be added by default though as probably not all need it.

But then media picker could in that case present the image instead of the icon. Actually I recall @madsrasmussen or @nielslyngsoe worked on a feature to customize the view for a specific media type? If not possible to change default view of Video media type, maybe a custom media type inheriting Video with an additional poster/cover property.

**Reply by bjarnef** [reply]
I was referring to the `umb-file-preview` https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/components/media/umbfilepreview/umb-file-preview.html

https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/common/services/mediapreview.service.js#L57-L71

https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/components/media/umbmediapreview/umbmediapreview.component.js#L26

Currently it is only used in `umb-media-preview` which is used here:
https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/components/upload/umb-property-file-upload.html#L22-L28

and here:
https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediaentryeditor/mediaentryeditor.html#L84-L93

**Reply by bjarnef** [reply]
Not sure if the `umb-media-preview` could be used here in `umb-media-card` which is the component media picker use:

https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/components/mediacard/umb-media-card.html#L16-L37

**Reply by martinlcp** [reply]
> Maybe Umbraco could have a special alias like `umbracoFilePoster`, `umbracoCover` or something else?
> 
> E.g. added to Video media type using a file upload as well:
> 
> ![image](https://user-images.githubusercontent.com/2919859/198567875-554638b1-abfc-4f7c-88b7-171025e7b4c4.png)
> 
> Not sure if this should be added by default though as probably not all need it.
> 
> But then media picker could in that case present the image instead of the icon. Actually I recall @madsrasmussen or @nielslyngsoe worked on a feature to customize the view for a specific media type? If not possible to change default view of Video media type, maybe a custom media type inheriting Video with an additional poster/cover property.

Yup that's exactly what i'm proposing, adding the poster property by default to a video Media type would mean it gets used in the picker. If you don't supply a poster frame it reverts back to the icon.

**Reply by bjarnef** [reply]
> Not sure if the `umb-media-preview` could be used here in `umb-media-card` which is the component media picker use:
> 
> https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/components/mediacard/umb-media-card.html#L16-L37

I think if the media card used the `umb-media-preview` component it may be possible to swap/replace the view with a custom view and handle the custom property added e.g. `poster`, `cover` or what the alias is named.

But I haven't looked further into this. Maybe @nielslyngsoe could answer this?

**Reply by skttl** [reply]
I don't know what HQs stance on these "secret" property aliases is, but I would like a standardized way of adding thumbnails to non-image things, like videos, files etc.

I think it's a great idea to add an `Upload field` with the alias `umbracoPoster` or similar (thumbnail, preview, cover etc.) and then that image would be used in media picker previews.

**Reply by bjarnef** [reply]
I think the `umb-file-preview` is based on file extension though and not media type.
Maybe this part should include a custom view/component https://github.com/umbraco/Umbraco-CMS/blob/v11/contrib/src/Umbraco.Web.UI.Client/src/views/components/mediacard/umb-media-card.html#L16-L37 like `umb-media-type-preview` - inside it could eventually fallback to `umb-file-preview`, but with the ability to define a custom view (maybe based on media type alias?).

For example `~/App_Plugins/MyPlugin/BackOffice/Views/MediaTypes/video.html` .. inside the video it has access to properties of the media type, so it could handle any custom properties.

Not sure how the path should be defined. Maybe search a specific path like custom icons ... or a setting on media type to select a custom thumbnail view (similar to custom view in a block list/grid block).
